mlir-tensorrt/executor/include/mlir-executor/Runtime/API/API.h

```diff
@@ -773,6 +773,9 @@ class AllocTracker {
   /// Return true if the tracker's map contains `ptr`.
   bool contains(uintptr_t ptr) const;
 
+  /// Report total CPU and GPU memory allocated by runtime client.
+  std::pair<int64_t, int64_t> reportAllocatedMemory() const;
```

Comment:
There actually could  are more  types than just these two, so I'd prefer if we separate it into a struct or array. Array could be indexed by all the potential values of `PointerType`.